src/js/game/GameController.js

```diff
@@ -1280,14 +1280,15 @@ class GameController {
   }
 
   placeBlock(commandQueueItem, blockType) {
-    let blockIndex = (this.levelModel.yToIndex(this.levelModel.player.position[1]) + this.levelModel.player.position[0]);
-    let blockTypeAtPosition = this.levelModel.actionPlane._data[blockIndex].blockType;
+    const position = this.levelModel.player.position;
+    let blockTypeAtPosition = this.levelModel.actionPlane.getBlockAt(position).blockType;
+
     if (this.levelModel.canPlaceBlock()) {
       if (blockTypeAtPosition !== "") {
-        this.levelModel.destroyBlock(blockIndex);
+        this.levelModel.destroyBlock(position);
```

Comment:
note that destroyBlock expects to be passed a position argument, so I'm pretty sure this wasn't working correctly before